driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionExternalKeyVaultTest.java

```diff
@@ -0,0 +1,170 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.AutoEncryptionSettings;
+import com.mongodb.ClientEncryptionSettings;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.MongoSecurityException;
+import com.mongodb.client.model.vault.EncryptOptions;
+import com.mongodb.client.vault.ClientEncryption;
+import com.mongodb.client.vault.ClientEncryptions;
+import org.bson.BsonBinary;
+import org.bson.BsonBinarySubType;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.UuidRepresentation;
+import org.bson.codecs.UuidCodec;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mongodb.ClusterFixture.isNotAtLeastJava8;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.Fixture.getMongoClientSettingsBuilder;
+import static org.bson.codecs.configuration.CodecRegistries.fromCodecs;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
+import static util.JsonPoweredTestHelper.getTestDocument;
+
+@RunWith(Parameterized.class)
+public class ClientSideEncryptionExternalKeyVaultTest {
+    private MongoClient client, clientEncrypted;
+    private ClientEncryption clientEncryption;
+    private final boolean withExternalKeyVault;
+
+    public ClientSideEncryptionExternalKeyVaultTest(final boolean withExternalKeyVault) {
+        this.withExternalKeyVault = withExternalKeyVault;
+    }
+
+    @Before
+    public void setUp() throws IOException, URISyntaxException {
+        assumeFalse(isNotAtLeastJava8());
+        assumeTrue(serverVersionAtLeast(4, 1));
+        assumeTrue("Encryption test with external keyVault is disabled",
+                System.getProperty("org.mongodb.test.awsAccessKeyId") != null
+                    && !System.getProperty("org.mongodb.test.awsAccessKeyId").isEmpty());
+
+        CodecRegistry codecRegistry = fromRegistries(
```

Comment:
I don't think we need a customer CodecRegistry and therefore, don't need this MongoClient at all.  Can just use the default.